src/modules/NtpClient.rb

```diff
@@ -1099,10 +1099,12 @@ def update_cfa_record(record)
     def write_ntp_conf
       records_for_write.each do |record|
         unless record["cfa_record"]
-          record["cfa_record"] = CFA::NtpConf::Record.record_class(record["type"]).new
-          ntp_conf.records << record["cfa_record"]
+          ntp_conf.records << CFA::NtpConf::Record.record_class(record["type"]).new
+          record["cfa_record"] = ntp_conf.records.last
```

Comment:
well, if you feel brave enough, you can try to write reasonable test for write, which verify that really modified file is written. Now it do a lot of stuff and I do not understand half of them.